### ISSUE TITLE
[cloud_firestore] fatal error: could not build module 'FirebaseFirestore'

### ISSUE THREAD

**Issue by SeanOThomas** [FirebaseExtended/flutterfire#2337]
**Describe the bug**

EDIT: I confirmed this is also an issue for firebase_storage: ^3.1.5

The `cloud_firestore` doesn't build in Xcode out-of-the-box. I see the following error: 

```
/Users/sthomas/Desktop/Android_Projects/nothing_new2/ios/Pods/FirebaseFirestore/Firestore/Source/Public/FIRFirestoreSettings.h:46:1: error: property with 'retain (or strong)' attribute must be of object type
    @property(nonatomic, strong) dispatch_queue_t dispatchQueue;
```
A workaround is to go into the files and remove the `strong` attribute. However, whenever I do a `pod install`, these files get overidden. So this solution is not ideal.  

**To Reproduce**
1. Start a new Flutter project with Swift enabled
2. Add `cloud_firestore: ^0.13.4+2` to `pubspec.yaml`
4. Build project in Xcode

**Expected behavior**
The project should build successfully. 

**Additional context**

Build output:

```
Xcode's output:
↳
    While building module 'FirebaseFirestore' imported from /Users/sthomas/Desktop/Android_Projects/nothing_new2/ios/Pods/Headers/Public/Firebase/Firebase.h:49:
    In file included from <module-includes>:1:
    In file included from /Users/sthomas/Desktop/Android_Projects/nothing_new2/ios/Pods/Target Support Files/FirebaseFirestore/FirebaseFirestore-umbrella.h:17:
    In file included from /Users/sthomas/Desktop/Android_Projects/nothing_new2/ios/Pods/FirebaseFirestore/Firestore/Source/Public/FirebaseFirestore.h:25:
    /Users/sthomas/Desktop/Android_Projects/nothing_new2/ios/Pods/FirebaseFirestore/Firestore/Source/Public/FIRFirestoreSettings.h:46:1: error: property with 'retain (or strong)' attribute must be of object type
    @property(nonatomic, strong) dispatch_queue_t dispatchQueue;
    ^
    1 error generated.
    In file included from /Users/sthomas/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_core-0.4.4+3/ios/Classes/FLTFirebaseCorePlugin.m:7:
    /Users/sthomas/Desktop/Android_Projects/nothing_new2/ios/Pods/Headers/Public/Firebase/Firebase.h:49:13: fatal error: could not build module 'FirebaseFirestore'
        #import <FirebaseFirestore/FirebaseFirestore.h>
         ~~~~~~~^
    2 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```

**Comment by SeanOThomas** [comment]
I confirmed this is also an issue for `  firebase_storage: ^3.1.5`

**Comment by TahaTesser** [comment]
Hi @SeanOThomas 
can you please provide your `flutter doctor -v` and `flutter run --verbose`? 
Also, to better address the issue, would be helpful if you could post a minimal code sample to reproduce the problem
Thank you

**Comment by Salakar** [comment]
Hey, as part of our on-going work for #2582, this should have been resolved in our Firebase Firestore rework (#2913) - which has now been merged into master. We'll look at publishing some prereleases in the next few days. Thank you
